base/math.jl

```diff
@@ -1308,14 +1326,37 @@ end
     return T(exp2(log2(abs(widen(x))) * y))
 end
 
-# compensated power by squaring
 @constprop :aggressive @inline function ^(x::Float64, n::Integer)
+    T = Int64
+    m = n >= typemax(T) ? typemax(T) : n <= typemin(T) ? typemin(T) : n
+    x^(m%Int64)
+end
+# compensated power by squaring
+@constprop :aggressive @inline function ^(x::Float64, n::Int64)
     n == 0 && return one(x)
-    return pow_body(x, n)
+    if use_power_by_squaring(n)
+        return pow_body(x, n)
+    else
+        s = ifelse(x < 0 && isodd(n), -1.0, 1.0)
```

Comment:
can't this just be `pow_body(x, Float64(n))`?

Reply:
I will check, if `pow_body(Float,Float)` supports negative `x`. In the original code comments there is a hint, that the sign bit of `x` is used for a special purpose.

Reply:
oh yeah, I think I used it for subnormal exponents